samcli/lib/init/cli_template_modifier.py

```diff
@@ -0,0 +1,188 @@
+"""
+Class to parse and update template when tracing is enabled
+"""
+import logging
+from yaml.parser import ParserError
+from samcli.yamlhelper import parse_yaml_file
+
+
+LOG = logging.getLogger(__name__)
+
+
+class TemplateModifier:
```

Comment:
Since this is a completely new piece of code, I'd encourage you to add type checking

Reply:
This class sounds like it could be used for any field to update/change template. But it is only working for `Tracing` at the moment. There is no generic way to extend this to support other fields at the moment. For that reason, I would consider change naming of this class specific to Tracing, and/or extend generic parts into separate class.